src/main/java/hudson/plugins/jira/JiraCreateIssueNotifier.java

```diff
@@ -451,30 +451,40 @@ public FormValidation doCheckProjectKey(@QueryParameter String value) throws IOE
             return FormValidation.ok();
         }
 
-        public ListBoxModel doFillPriorityIdItems() {
+        public ListBoxModel doFillPriorityIdItems(@AncestorInPath Item item) {
+            Project project = null;
             ListBoxModel items = new ListBoxModel().add(""); // optional field
-            for (JiraSite site : JiraGlobalConfiguration.get().getSites()) {
-                JiraSession session = site.getSession();
-                if (session != null) {
-                    for (Priority priority : session.getPriorities()) {
-                        items.add("[" + site.getName() + "] " + priority.getName(), String.valueOf(priority.getId()));
+            if(item instanceof Project) {
+                project = (Project) item;
+                final JiraProjectProperty jiraProjectProperty = (JiraProjectProperty) project.getProperty(JiraProjectProperty.class);
+                JiraSite site = jiraProjectProperty.getSite();
+                if(null != site) {
+                    JiraSession session = site.getSession();
+                    if (session != null) {
+                        for (Priority priority : session.getPriorities()) {
+                            items.add("[" + site.getName() + "] " + priority.getName(), String.valueOf(priority.getId()));
+                        }
                     }
                 }
             }
             return items;
         }
 
-        public ListBoxModel doFillTypeIdItems() {
```

Comment:
Can you add a short description for this method as it became quite long now? One-liner will be enough :)